src/coreclr/src/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -2619,32 +2619,32 @@ public override EventInfo[] GetEvents(BindingFlags bindingAttr)
         }
 
         [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicFields | DynamicallyAccessedMemberTypes.NonPublicFields)]
-        public override FieldInfo[] GetFields(BindingFlags bindingAttr)
+        public override RuntimeFieldInfo[] GetFields(BindingFlags bindingAttr)
         {
             return GetFieldCandidates(null, bindingAttr, false).ToArray();
         }
 
-        public override Type[] GetInterfaces()
+        public override RuntimeType[] GetInterfaces()
```

Comment:
Note that this is potential breaking change if the caller mutates the array and tries to store non-RuntimeType types into the array.